File: demo-service-impl/src/main/java/com/github/zubmike/service/demo/logic/DictionaryLogic.java
package com.github.zubmike.service.demo.logic;

import com.github.zubmike.core.types.BasicDictItem;
import com.github.zubmike.core.utils.DictItemUtils;
import com.github.zubmike.service.demo.dao.PlanetarySystemDao;

import javax.inject.Inject;
import java.util.List;

public class DictionaryLogic {

	private final PlanetarySystemDao planetarySystemDao;

	@Inject
	public DictionaryLogic(PlanetarySystemDao planetarySystemDao) {
		this.planetarySystemDao = planetarySystemDao;
	}

	public List<BasicDictItem> getPlanetarySystems() {
		return DictItemUtils.createBasicDictItems(planetarySystemDao.getAll());
	}

}

File: demo-service-kotlin-impl/src/main/kotlin/com/github/zubmike/service/demo/logic/DictionaryLogic.kt
package com.github.zubmike.service.demo.logic

import com.github.zubmike.core.types.BasicDictItem
import com.github.zubmike.core.types.DictItem
import com.github.zubmike.service.demo.dao.PlanetarySystemDao

class DictionaryLogic(
        private val planetarySystemDao: PlanetarySystemDao) {

    fun getPlanetarySystems(): List<BasicDictItem> {
        return planetarySystemDao.all
                .map { createBasicDictItem(it) }
    }

    private fun createBasicDictItem(item: DictItem<Int>): BasicDictItem {
        return BasicDictItem(item.id, item.name)
    }

}